src/utils/Configuration.ts

```diff
@@ -1,98 +1,130 @@
+   * @returns Configuration
+   */
+  public static getInstance(): Configuration {
+    if (!this.instance) {
+      this.instance = new Configuration("../config/config.yml");
     }
 
-    /**
-     * Retrieves the entire config as an object
-     * @returns any
-     */
-    public getConfig(): any {
-        return this.config;
+    return Configuration.instance;
+  }
+
+  /**
+   * Retrieves the DynamoDB config
+   * @returns IDynamoDBConfig
+   */
+  public getS3Config(): IS3Config {
+    if (!this.config.s3) {
+      throw new Error(ERRORS.DYNAMO_DB_CONFIG_NOT_DEFINED);
     }
 
-    /**
-     * Retrieves the DynamoDB config
-     * @returns IDynamoDBConfig
-     */
-    public getS3Config(): IS3Config {
-        if (!this.config.s3) {
-            throw new Error(ERRORS.DYNAMO_DB_CONFIG_NOT_DEFINED);
-        }
+    // Not defining BRANCH will default to local
+    const env: string = (!process.env.BRANCH || process.env.BRANCH === "local") ? "local" : "remote";
 
-        // Not defining BRANCH will default to local
-        const env: string = (!process.env.BRANCH || process.env.BRANCH === "local") ? "local" : "remote";
+    return this.config.s3[env];
+  }
 
-        return this.config.s3[env];
+  /**
+   * Retrieves the Lambda Invoke config
+   * @returns IInvokeConfig
+   */
+  public getInvokeConfig(): IInvokeConfig {
+    if (!this.config.invoke) {
+      throw new Error("Lambda Invoke config is not defined in the config file.");
     }
 
-    /**
-     * Retrieves the Lambda Invoke config
-     * @returns IInvokeConfig
-     */
-    public getInvokeConfig(): IInvokeConfig {
-        if (!this.config.invoke) {
-            throw new Error("Lambda Invoke config is not defined in the config file.");
-        }
+    // Not defining BRANCH will default to local
+    const env: string = (!process.env.BRANCH || process.env.BRANCH === "local") ? "local" : "remote";
 
-        // Not defining BRANCH will default to local
-        const env: string = (!process.env.BRANCH || process.env.BRANCH === "local") ? "local" : "remote";
+    return this.config.invoke[env];
+  }
 
-        return this.config.invoke[env];
+  /**
+   * Retrieves the Gov Notify config
+   * @returns IMOTConfig
+   */
+  public async getGovNotifyConfig(): Promise<IMOTConfig> {
+    if (!this.config.notify) {
+      throw new Error(ERRORS.MOT_CONFIG_NOT_DEFINED);
     }
-
-    /**
-     * Retrieves the Gov Notify config
-     * @returns IMOTConfig
-     */
-    public getGovNotifyConfig(): IMOTConfig {
-        if (!this.config.notify) {
-            throw new Error(ERRORS.MOT_CONFIG_NOT_DEFINED);
-        }
-
-        return this.config.notify;
+    if (!this.config.notify.api_key || !this.config.notify.endpoint) {
+      await this.setSecrets();
     }
-
+    return this.config.notify;
+  }
+
+  /**
+   * Sets the secrets needed to use GovNotify
+   * @returns Promise<void>
+   */
+  private async setSecrets(): Promise<void> {
```

Comment:
Slightly confusing name. It's "setting" them locally, but actually getting them from remote? Maybe "fetchSecrets"?

Reply:
Well, it can 'fetch' secrets from secret manager or local file, so I decided on 'set'